lib/airbrake-ruby/thread_context.rb

```diff
@@ -0,0 +1,56 @@
+module Airbrake
+  ##
+  # A class that is responsible for storing Airbrake Ruby thread context. A
+  # thread context is a hash of arbitrary objects to be sent to Airbrake. The
+  # implementation of the class relies on thread local variables.
+  #
+  # @since 2.1.0
+  class ThreadContext
+    ##
+    # @return [String] the prefix of an Airbrake Ruby thread local variable
+    CONTEXT_PREFIX = 'airbrake_ruby_context'.freeze
+
+    def initialize(name)
+      @context_key = "#{CONTEXT_PREFIX}_#{name}"
+      clear
```

Comment:
Every notifier will have its own space. This is get constructed when a Notifier is initialised. We call `clear` here to set the initial hash. The same notifier can't be initialised twice, so this should be safe, unless I missed something.